src/sparsezoo/objects/model.py

```diff
@@ -887,6 +972,73 @@ def download_framework_files(
             downloaded_paths.append(file.path)
         return downloaded_paths
 
+    def download_checkpoints_folders(
```

Comment:
let's clean up this function a bit -- it's very nested and very long meaning there are a lot of complicated flows in this that are hard to read through